src/System/Process/Typed.hs

```diff
@@ -624,7 +618,17 @@ startProcess pConfig'@ProcessConfig {..} = liftIO $ do
 
     pExitCode <- newEmptyTMVarIO
     waitingThread <- async $ do
-        ec <- P.waitForProcess pHandle
+        ec <-
+          if multiThreadedRuntime
```

Comment:
We probably want to mention in the haddocks that this function works on non-threaded but that it may be slower than desired dure to the polling (and also the Windows issue).

Reply:
I've added a comment to the `README`. I don't think the Haddocks are the right place for this, because the inefficiency affects basically the entire API. (Note that the module refers to the README.md for its documentations.)

Reply:
Makes sense!